src/libraries/Native/Windows/System.IO.Compression.Native/CMakeLists.txt

```diff
@@ -24,9 +24,10 @@ if(${CLR_CMAKE_HOST_ARCH} STREQUAL arm OR ${CLR_CMAKE_HOST_ARCH} STREQUAL arm64)
         zlib-ng/deflate_slow.c
         zlib-ng/deflate.c
         zlib-ng/functable.c
-        zlib-ng/gzlib.c
-        zlib-ng/gzread.c
-        zlib-ng/gzwrite.c
+        # Exclude files to fix error LNK2019: unresolved external symbol 'gzFile' open referenced in functions gz_open, gz_close, gz_load, gz_comp
```

Comment:
Should find out why it is giving that linker error. If dotnet does not use any of the gz_* functions then you don’t need those sources and can also set -DWITH_GZFILEOP=0.

Reply:
Why not just use add_subdirectory to add our CMake project?

Reply:
I'm looking into this again now that we're back from the Thanksgiving holidays.

I'm exploring the use of `add_subdirectory` and have some questions, but I'd like to first rebase this PR with the latest bits in main, because there was a big refactoring of these folders. I'll ask you the questions when I refresh the PR.

Reply:
Sounds good.

Reply:
> first rebase this PR with the latest bits in main

Maybe you can put zlib-ng under src/native/external as part of this rebase as discussed in 
 https://github.com/dotnet/runtime/pull/61883#issuecomment-978815762, so that we do not have to move it later?